examples/publisher_untyped.rs

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+use iceoryx_rs::PublisherBuilder;
+use iceoryx_rs::Runtime;
+
+use bytes::BufMut;
+
+use std::error::Error;
+use std::thread;
+use std::time::Duration;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    Runtime::init("publisher_simple");
+
+    let publisher = PublisherBuilder::<[u8]>::new("Radar", "FrontLeft", "Counter").create()?;
+
+    let mut counter = 0u32;
+    loop {
+        let mut sample = publisher.loan(std::mem::size_of::<u32>(), std::mem::align_of::<u32>())?;
+        sample.as_mut().put_u32(counter);
+        publisher.publish(sample);
```

Comment:
@neilisaac does this fit your use case?

I think I will introduce a `loan_uninitialized` and let `loan` zero the memory.

Reply:
Yes, this looks great!

For reference, what I had was:
```rust
pub struct PubChunkHandle<'a> {
    data: *mut ffi::c_void,
    publisher: &'a Publisher,
    sent: bool,  // used in Drop
    size: usize,
}

impl PubChunkHandle<'_> {
    /// Send the data via the associated Publisher
    pub fn publish(mut self) {
        unsafe { sys::iox_pub_publish_chunk(self.publisher.0, self.data) };
        self.sent = true;
    }

    /// Safe interface to write into a shared memory chunk.
    pub fn as_uninit_slice(&mut self) -> &mut UninitSlice {
        unsafe { UninitSlice::from_raw_parts_mut(self.data.cast(), self.size) }
    }

    /// Expose the payload portion of the chunk as a mutable byte slice.
    /// This can be written to conveniently using bytes::BufMut.
    /// # Safety
    /// Despite this function not being marked as unsafe, the data returned by as_slice is uninitialized.
    pub fn as_slice(&mut self) -> &mut [u8] {
        unsafe { std::slice::from_raw_parts_mut(self.data.cast(), self.size) }
    }

    /// Cast the chunk to a given type T.
    /// This may be useful with a #[repr(C)] type for example.
    /// # Safety
    /// This function is marked as unsafe because it is the caller's responsibility
    /// to ensure that T can safely be written into the chunk.
    /// T must fit in the allocated chunk and must not reference data outside the chunk (ex. via pointers/references)
    pub unsafe fn as_uninitialized<T>(&mut self) -> &mut std::mem::MaybeUninit<T> {
        &mut *self.data.cast()
    }
}
```

The main design difference there I think is the PubChunkHandle borrows the Publisher, but otherwise very similar.

Reply:
Nice. Currently I plan to have this methods
```rust
pub fn loan_zeroed_slice(&self, len: usize) -> Result<SampleMut<[u8]>, IceoryxError> {
    unimplemented!()
}

pub fn loan_aligned_zeroed_slice(&self, len: usize, alignment: usize) -> Result<SampleMut<[u8]>, IceoryxError> {
    unimplemented!()
}

pub fn loan_uninit_slice(&self, len: usize) -> Result<SampleMut<[MaybeUninit<u8>]>, IceoryxError> {
    unimplemented!()
}

pub fn loan_aligned_uninit_slice(&self, len: usize, alignment: usize) -> Result<SampleMut<[MaybeUninit<u8>]>, IceoryxError> {
    unimplemented!()
}
```
I have to check the ergonomics with `[MaybeUninit<u8>]`. Maybe this needs to be done in a different way.

It would also be possible to not restrict the API to `[u8]` slices but to allow all types with `T: ShmSend`. With this one can have dynamic data at runtime. At least the most simple one.

@elfenpiff what do you think about having a specialization `Sample<cxx::span<T>>` for the typed C++ API?